File: api/MfaApi/src/Modules/Exchange/Contracts/GetExchangesResponse.cs
namespace MfaApi.Modules.Exchange;

public class GetExchangesResponse {
    public required Guid Id { get; set; }
    public required int Year { get; set; }
    public required ExchangeType ExchangeType { get; set; }
    public required Guid MemberId { get; set; }
    public MemberDto? Member { get; set; }

    public class MemberDto {
        public required Guid Id { get; set; }
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
    }
}

File: dashboard/src/modules/exchange/types/get-exchanges-response.ts
import { ExchangeType } from "./exchange-type";

export interface GetExchangesResponse {
  id: string;
  year: number;
  exchangeType: ExchangeType;
  memberId: string;
  member: {
    id: string;
    firstName: string;
    lastName: string;
  } | null;
}